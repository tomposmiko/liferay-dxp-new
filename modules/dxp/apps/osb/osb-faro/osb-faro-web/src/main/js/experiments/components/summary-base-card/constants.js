export const CLASSNAME = 'analytics-summary-card';
