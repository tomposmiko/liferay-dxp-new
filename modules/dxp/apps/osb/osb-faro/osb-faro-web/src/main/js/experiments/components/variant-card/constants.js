export const CLASSNAME = 'analytics-variant-card';
export const CONTROL_COLOR = '#6B6C7E';
