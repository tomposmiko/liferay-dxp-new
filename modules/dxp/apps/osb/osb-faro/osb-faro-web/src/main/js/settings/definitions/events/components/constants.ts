export const LIMIT_REACHED_ALERT_ID = 'limitReachedAlertId';
