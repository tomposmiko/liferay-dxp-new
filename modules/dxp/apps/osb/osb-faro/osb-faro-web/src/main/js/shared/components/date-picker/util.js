import moment from 'moment';
import {conformsTo, isNil} from 'lodash';

/**
 * Represents a time between two dates.
 * @typedef Range
 * @property {Moment} start - The start of the range.
 * @property {Moment} end - The end of the range.
 */

export function isAboveMaxRange(date, maxRange) {
	return (
		isRange(date) &&
		date.start &&
		date.end &&
		maxRange &&
		moment
			.duration({
				from: date.start,
				to: date.end
			})
			.asDays() > maxRange
	);
}

/**
 * A validation function that determines if a value is a valid
 * moment instance, allowing for null.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
function isMomentOrNil(value) {
	return isNil(value) || moment.isMoment(value);
}

/**
 * A validation function that determines if the value is a valid
 * moment or range.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
export function isDateOrRange(value) {
	return isNil(value) || moment.isMoment(value) || isRange(value);
}

/**
 * Determines if a date is in a {@link Range}.
 * @param {Range|Moment} dateOrRange - The range to check in.
 * @param {Moment} date - The date to check for.
 * @returns {boolean}
 */
export function isInRange(dateOrRange, date) {
	if (moment.isMoment(dateOrRange)) {
		return false;
	}

	const {end, start} = dateOrRange;

	return start && date.isAfter(start) && end && date.isBefore(end);
}

/**
 * A validation function that determines if a value is a valid {@link Range}.
 * @param {Object} value - The value to test.
 * @returns {boolean}
 */
export function isRange(value) {
	return conformsTo(value, {
		end: isMomentOrNil,
		start: isMomentOrNil
	});
}

/**
 * Adds a date to an existing {@link Range}.
 *
 * This function will try to be smart and first update the
 * start or end if either are null. If both exist, then it will return
 * a new range with the start set to the new date, and the end reset
 * to null.
 * @param {Range} range - The range to update.
 * @param {Moment} date - The date to add to the range.
 * @returns {Range}
 */
export function updateRange(range, date) {
	if (!range.start) {
		return {
			...range,
			start: date
		};
	}

	if (range.start && !range.end) {
		if (date.isSameOrAfter(range.start)) {
			return {
				...range,
				end: date
			};
		} else {
			return {
				end: range.start,
				start: date
			};
		}
	}

	if (date.isSame(range.start) && date.isSame(range.end)) {
		return range;
	}

	return {
		end: null,
		start: date
	};
}
