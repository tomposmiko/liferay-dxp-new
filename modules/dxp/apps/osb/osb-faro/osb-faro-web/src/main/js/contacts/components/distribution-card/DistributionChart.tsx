import autobind from 'autobind-decorator';
import Button from 'shared/components/Button';
import Card from 'shared/components/Card';
import ErrorDisplay from 'shared/components/ErrorDisplay';
import Promise from 'metal-promise';
import React from 'react';
import Spinner from 'shared/components/Spinner';
import {ANIMATION_DURATION, AXIS, getTextWidth} from 'shared/util/recharts';
import {autoCancel, hasRequest} from 'shared/util/request-decorator';
import {
	Bar,
	CartesianGrid,
	Cell,
	ComposedChart,
	ResponsiveContainer,
	XAxis,
	YAxis
} from 'recharts';
import {connect, ConnectedProps} from 'react-redux';
import {DistributionTab} from 'shared/util/records';
import {FieldTypes} from 'shared/util/constants';
import {getBarColor} from 'shared/util/charts';
import {hasChanges} from 'shared/util/react';
import {List, Map} from 'immutable';
import {noop, pickBy} from 'lodash';
import {RootState} from 'shared/store';

const BAR_WIDTH = 30;
const CHART_DATA_ID = 'count';
const CHART_PADDING = 60;
const MAX_BARS = 10;

const connector = connect(
	(state: RootState, {distributionKey}: {distributionKey: string}) => {
		const distributionIMap = state.getIn(
			['distributions', distributionKey],
			Map()
		);

		return {
			error: distributionIMap.get('error', false),
			individualFieldDistributionIList: distributionIMap
				.getIn(['data', 'items'], List())
				.slice(0, 11),
			loading: distributionIMap.get('loading', true)
		};
	}
);

type PropsFromRedux = ConnectedProps<typeof connector>;

interface IDistributionChartProps extends PropsFromRedux {
	channelId: string;
	distributionKey: string;
	error: boolean;
	fetchDistribution: (params: object) => typeof Promise;
	groupId: string;
	id: string;
	individualFieldDistributionIList: List<Map<string, any>>;
	loading: boolean;
	noResultsRenderer?: () => React.ReactElement;
	selectedTab: DistributionTab;
	viewAllLink: string;
}

@hasRequest
class DistributionChart extends React.Component<
	IDistributionChartProps,
	{hoverIndex: number}
> {
	state = {
		hoverIndex: -1
	};

	componentDidMount() {
		this.handleFetchChartData();
	}

	componentDidUpdate(prevProps) {
		if (hasChanges(prevProps, this.props, 'selectedTab')) {
			this.handleFetchChartData();
		}
	}

	formatChartData(fieldDistributions, histogram) {
		return fieldDistributions.map(({count, values}) => ({
			count,
			graphValue: histogram ? (values[0] + values[1]) / 2 : values[0],
			values
		}));
	}

	getYAxisTicks(fieldDistributions, histogram) {
		return [
			...fieldDistributions.map(item => item.values[0]),
			histogram &&
				fieldDistributions.length &&
				fieldDistributions[fieldDistributions.length - 1].values[1]
		].filter(Boolean);
	}

	@autoCancel
	@autobind
	handleFetchChartData() {
		const {
			channelId,
			fetchDistribution,
			groupId,
			id,
			selectedTab: {context, numberOfBins, propertyId}
		} = this.props;

		return fetchDistribution(
			pickBy({
				channelId,
				context,
				count: MAX_BARS,
				fieldMappingFieldName: propertyId,
				groupId,
				id,
				individualSegmentId: id,
				numberOfBins
			})
		).catch(noop);
	}

	render() {
		const {
			props: {
				error,
				individualFieldDistributionIList,
				loading,
				noResultsRenderer,
				selectedTab: {propertyType},
				viewAllLink
			},
			state: {hoverIndex}
		} = this;

		const individualFieldDistribution =
			individualFieldDistributionIList.toJS();

		const histogram = propertyType === FieldTypes.Number;

		const yAxisTicks = this.getYAxisTicks(
			individualFieldDistribution,
			histogram
		);

		const formattedChartData = this.formatChartData(
			individualFieldDistribution,
			histogram
		);

		const fieldDistributionsCount = individualFieldDistribution.length;

		const yAxisDomain = histogram
			? [yAxisTicks[0], yAxisTicks[yAxisTicks.length - 1]]
			: [0, 'auto'];

		const yAxisWidth = yAxisTicks.reduce((acc, item) => {
			const textWidth = getTextWidth(item.toString());

			return textWidth > acc ? textWidth : acc;
		}, 60);

		return (
			<>
				<Card.Body alignCenter>
					{error && (
						<ErrorDisplay
							onReload={this.handleFetchChartData}
							spacer
						/>
					)}

					{loading && <Spinner spacer />}

					{!error && !loading && (
						<>
							{!fieldDistributionsCount &&
								noResultsRenderer &&
								noResultsRenderer()}

							{!!fieldDistributionsCount && (
								<ResponsiveContainer
									height={
										BAR_WIDTH * MAX_BARS + CHART_PADDING
									}
								>
									<ComposedChart
										data={formattedChartData}
										layout='vertical'
									>
										<CartesianGrid
											horizontal={false}
											stroke={AXIS.gridStroke}
											strokeDasharray='3 3'
										/>

										<YAxis
											axisLine={{
												stroke: AXIS.borderStroke
											}}
											dataKey='graphValue'
											domain={yAxisDomain}
											tickFormatter={val => val}
											ticks={yAxisTicks}
											type={
												histogram
													? 'number'
													: 'category'
											}
											width={yAxisWidth}
										/>

										<YAxis
											axisLine={{
												stroke: AXIS.borderStroke
											}}
											dataKey='graphValue'
											domain={yAxisDomain}
											orientation='right'
											tick={false}
											tickLine={false}
											yAxisId='right'
										/>

										<XAxis
											axisLine={{
												stroke: AXIS.borderStroke
											}}
											dataKey={CHART_DATA_ID}
											interval='preserveStart'
											orientation='top'
											scale='linear'
											tickLine={false}
											type='number'
										/>

										<XAxis
											axisLine={{
												stroke: AXIS.borderStroke
											}}
											dataKey={CHART_DATA_ID}
											tick={false}
											tickLine={false}
											xAxisId='bottom'
										/>

										<Bar
											animationDuration={
												ANIMATION_DURATION.bar
											}
											dataKey={CHART_DATA_ID}
											onMouseEnter={(e, index) =>
												this.setState({
													hoverIndex: index
												})
											}
											onMouseLeave={() =>
												this.setState({
													hoverIndex: -1
												})
											}
										>
											{formattedChartData.map(
												(item, index) => (
													<Cell
														fill={getBarColor(
															index,
															hoverIndex
														)}
														key={`cell-${index}`}
													/>
												)
											)}
										</Bar>
									</ComposedChart>
								</ResponsiveContainer>
							)}
						</>
					)}
				</Card.Body>

				<Card.Footer>
					<Button
						display='link'
						href={viewAllLink}
						icon='angle-right'
						iconAlignment='right'
						size='sm'
					>
						{Liferay.Language.get('explore-breakdown')}
					</Button>
				</Card.Footer>
			</>
		);
	}
}

export default connector(DistributionChart);
