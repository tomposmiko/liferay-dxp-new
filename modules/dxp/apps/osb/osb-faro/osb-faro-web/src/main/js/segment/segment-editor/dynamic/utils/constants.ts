import {TimeSpans} from 'shared/util/constants';

export const EVER = 'ever';
export const SINCE = 'since';

export const isKnown = 'is-known';
export const isUnknown = 'is-unknown';

/**
 * Constants for date formatting
 */

export const INPUT_DATE_FORMAT = 'YYYY-MM-DD';
export const INPUT_DATE_TIME_FORMAT = 'YYYY-MM-DDTHH:mmZ';
export const INPUT_DISPLAY_DATE_TIME_FORMAT = 'YYYY-MM-DD HH:mm';

/**
 * Constants for OData query.
 */

export enum Conjunctions {
	And = 'and',
	Or = 'or'
}

export enum CustomFunctionOperators {
	AccountsFilter = 'accounts-filter',
	AccountsFilterByCount = 'accounts-filter-by-count',
	ActivitiesFilter = 'activities-filter',
	ActivitiesFilterByCount = 'activities-filter-by-count',
	EventsFilterByCount = 'events-filter-by-count',
	InterestsFilter = 'interests-filter',
	OrganizationsFilter = 'organizations-filter',
	SessionsFilter = 'sessions-filter'
}

export enum DisplayOnlyOperators {
	IsKnown = 'ne',
	IsUnknown = 'eq'
}

export enum FunctionalOperators {
	Between = 'between',
	Contains = 'contains'
}

export enum NotOperators {
	NotAccountsFilter = 'not-accounts-filter',
	NotAccountsFilterByCount = 'not-accounts-filter-by-count',
	NotActivitiesFilter = 'not-activities-filter',
	NotActivitiesFilterByCount = 'not-activities-filter-by-count',
	NotContains = 'not-contains',
	NotEventsFilterByCount = 'not-events-filter-by-count',
	NotOrganizationsFilter = 'not-organizations-filter',
	NotSessionsFilter = 'not-sessions-filter'
}

export const GROUP = 'GROUP';

export enum RelationalOperators {
	EQ = 'eq',
	GE = 'ge',
	GT = 'gt',
	LE = 'le',
	LT = 'lt',
	NE = 'ne'
}

/**
 * Constants to match property types in the passed in supportedProperties array.
 */

export enum PropertyTypes {
	AccountNumber = 'account-number',
	AccountText = 'account-text',
	Behavior = 'behavior',
	Boolean = 'boolean',
	Date = 'date',
	DateTime = 'date-time',
	Duration = 'duration',
	Event = 'event',
	Interest = 'interest',
	Number = 'number',
	OrganizationBoolean = 'organization-boolean',
	OrganizationDate = 'organization-date',
	OrganizationDateTime = 'organization-date-time',
	OrganizationNumber = 'organization-number',
	OrganizationSelectText = 'organization-select-text',
	OrganizationText = 'organization-text',
	SelectText = 'select-text',
	SessionDateTime = 'session-date-time',
	SessionGeolocation = 'session-geolocation',
	SessionNumber = 'session-number',
	SessionText = 'session-text',
	Text = 'text'
}

/**
 * Constants for CriteriaBuilder component.
 */

export const CUSTOM_FUNCTION_OPERATOR_KEY_MAP = {
	['accounts.filter']: CustomFunctionOperators.AccountsFilter,
	['accounts.filterByCount']: CustomFunctionOperators.AccountsFilterByCount,
	['activities.filter']: CustomFunctionOperators.ActivitiesFilter,
	['activities.filterByCount']:
		CustomFunctionOperators.ActivitiesFilterByCount,
	['events.filterByCount']: CustomFunctionOperators.EventsFilterByCount,
	['interests.filter']: CustomFunctionOperators.InterestsFilter,
	['organizations.filter']: CustomFunctionOperators.OrganizationsFilter,
	['sessions.filter']: CustomFunctionOperators.SessionsFilter
};

export const SUPPORTED_CONJUNCTION_OPTIONS = [
	{
		key: Conjunctions.And,
		label: Liferay.Language.get('and'),
		name: Conjunctions.And
	},
	{
		key: Conjunctions.Or,
		label: Liferay.Language.get('or'),
		name: Conjunctions.Or
	}
];

export const SUPPORTED_OPERATORS_MAP = {
	[PropertyTypes.AccountNumber]: [
		{
			key: CustomFunctionOperators.AccountsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.AccountsFilter
		}
	],
	[PropertyTypes.AccountText]: [
		{
			key: CustomFunctionOperators.AccountsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.AccountsFilter
		}
	],
	[PropertyTypes.Behavior]: [
		{
			key: CustomFunctionOperators.ActivitiesFilterByCount,
			label: Liferay.Language.get('has-fragment'),
			name: CustomFunctionOperators.ActivitiesFilterByCount
		},
		{
			key: NotOperators.NotActivitiesFilterByCount,
			label: Liferay.Language.get('has-not-fragment'),
			name: NotOperators.NotActivitiesFilterByCount
		}
	],
	[PropertyTypes.Boolean]: [
		{
			key: RelationalOperators.EQ,
			label: Liferay.Language.get('is-fragment'),
			name: RelationalOperators.EQ
		}
	],
	[PropertyTypes.Date]: [
		{
			key: RelationalOperators.LT,
			label: Liferay.Language.get('is-before-fragment'),
			name: RelationalOperators.LT
		},
		{
			key: RelationalOperators.EQ,
			label: Liferay.Language.get('is-fragment'),
			name: RelationalOperators.EQ
		},
		{
			key: RelationalOperators.GT,
			label: Liferay.Language.get('is-after-fragment'),
			name: RelationalOperators.GT
		}
	],
	[PropertyTypes.DateTime]: [
		{
			key: RelationalOperators.LT,
			label: Liferay.Language.get('is-before-fragment'),
			name: RelationalOperators.LT
		},
		{
			key: RelationalOperators.EQ,
			label: Liferay.Language.get('is-fragment'),
			name: RelationalOperators.EQ
		},
		{
			key: RelationalOperators.GT,
			label: Liferay.Language.get('is-after-fragment'),
			name: RelationalOperators.GT
		}
	],
	[PropertyTypes.Duration]: [
		{
			key: RelationalOperators.GT,
			label: Liferay.Language.get('greater-than-fragment'),
			name: RelationalOperators.GT
		},
		{
			key: RelationalOperators.LT,
			label: Liferay.Language.get('less-than-fragment'),
			name: RelationalOperators.LT
		}
	],
	[PropertyTypes.Event]: [
		{
			key: CustomFunctionOperators.EventsFilterByCount,
			label: Liferay.Language.get('has-fragment'),
			name: CustomFunctionOperators.EventsFilterByCount
		},
		{
			key: NotOperators.NotActivitiesFilterByCount,
			label: Liferay.Language.get('has-not-fragment'),
			name: NotOperators.NotActivitiesFilterByCount
		}
	],
	[PropertyTypes.Interest]: [
		{
			key: CustomFunctionOperators.InterestsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.InterestsFilter
		}
	],
	[PropertyTypes.Number]: [
		{
			key: RelationalOperators.EQ,
			label: Liferay.Language.get('is-equal-to-fragment'),
			name: RelationalOperators.EQ
		},
		{
			key: RelationalOperators.GT,
			label: Liferay.Language.get('greater-than-fragment'),
			name: RelationalOperators.GT
		},
		{
			key: RelationalOperators.LT,
			label: Liferay.Language.get('less-than-fragment'),
			name: RelationalOperators.LT
		},
		{
			key: RelationalOperators.NE,
			label: Liferay.Language.get('is-not-equal-to-fragment'),
			name: RelationalOperators.NE
		},
		{
			key: isKnown,
			label: Liferay.Language.get('is-known-fragment'),
			name: DisplayOnlyOperators.IsKnown
		},
		{
			key: isUnknown,
			label: Liferay.Language.get('is-unknown-fragment'),
			name: DisplayOnlyOperators.IsUnknown
		}
	],
	[PropertyTypes.OrganizationBoolean]: [
		{
			key: CustomFunctionOperators.OrganizationsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.OrganizationsFilter
		}
	],
	[PropertyTypes.OrganizationDate]: [
		{
			key: CustomFunctionOperators.OrganizationsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.OrganizationsFilter
		}
	],
	[PropertyTypes.OrganizationDateTime]: [
		{
			key: CustomFunctionOperators.OrganizationsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.OrganizationsFilter
		}
	],
	[PropertyTypes.OrganizationNumber]: [
		{
			key: CustomFunctionOperators.OrganizationsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.OrganizationsFilter
		}
	],
	[PropertyTypes.OrganizationSelectText]: [
		{
			key: CustomFunctionOperators.OrganizationsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.OrganizationsFilter
		},
		{
			key: NotOperators.NotOrganizationsFilter,
			label: Liferay.Language.get('is-not-fragment'),
			name: NotOperators.NotOrganizationsFilter
		}
	],
	[PropertyTypes.OrganizationText]: [
		{
			key: CustomFunctionOperators.OrganizationsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.OrganizationsFilter
		}
	],
	[PropertyTypes.SelectText]: [
		{
			key: RelationalOperators.EQ,
			label: Liferay.Language.get('is-fragment'),
			name: RelationalOperators.EQ
		},
		{
			key: RelationalOperators.NE,
			label: Liferay.Language.get('is-not-fragment'),
			name: RelationalOperators.NE
		}
	],
	[PropertyTypes.SessionDateTime]: [
		{
			key: CustomFunctionOperators.SessionsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.SessionsFilter
		}
	],
	[PropertyTypes.SessionGeolocation]: [
		{
			key: CustomFunctionOperators.SessionsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.SessionsFilter
		}
	],
	[PropertyTypes.SessionNumber]: [
		{
			key: CustomFunctionOperators.SessionsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.SessionsFilter
		}
	],
	[PropertyTypes.SessionText]: [
		{
			key: CustomFunctionOperators.SessionsFilter,
			label: Liferay.Language.get('is-fragment'),
			name: CustomFunctionOperators.SessionsFilter
		}
	],
	[PropertyTypes.Text]: [
		{
			key: RelationalOperators.EQ,
			label: Liferay.Language.get('is-fragment'),
			name: RelationalOperators.EQ
		},
		{
			key: RelationalOperators.NE,
			label: Liferay.Language.get('is-not-fragment'),
			name: RelationalOperators.NE
		},
		{
			key: FunctionalOperators.Contains,
			label: Liferay.Language.get('contains-fragment'),
			name: FunctionalOperators.Contains
		},
		{
			key: NotOperators.NotContains,
			label: Liferay.Language.get('does-not-contain-fragment'),
			name: NotOperators.NotContains
		},
		{
			key: isKnown,
			label: Liferay.Language.get('is-known-fragment'),
			name: DisplayOnlyOperators.IsKnown
		},
		{
			key: isUnknown,
			label: Liferay.Language.get('is-unknown-fragment'),
			name: DisplayOnlyOperators.IsUnknown
		}
	]
};

export const SUPPORTED_PROPERTY_TYPES_MAP = {
	[PropertyTypes.AccountNumber]: [CustomFunctionOperators.AccountsFilter],
	[PropertyTypes.AccountText]: [CustomFunctionOperators.AccountsFilter],
	[PropertyTypes.Behavior]: [
		CustomFunctionOperators.ActivitiesFilterByCount,
		NotOperators.NotActivitiesFilterByCount
	],
	[PropertyTypes.Boolean]: [RelationalOperators.EQ],
	[PropertyTypes.Date]: [
		RelationalOperators.EQ,
		RelationalOperators.GE,
		RelationalOperators.GT,
		RelationalOperators.LE,
		RelationalOperators.LT,
		RelationalOperators.NE
	],
	[PropertyTypes.DateTime]: [
		RelationalOperators.EQ,
		RelationalOperators.GE,
		RelationalOperators.GT,
		RelationalOperators.LE,
		RelationalOperators.LT,
		RelationalOperators.NE
	],
	[PropertyTypes.Duration]: [RelationalOperators.GT, RelationalOperators.LT],
	[PropertyTypes.Event]: [
		CustomFunctionOperators.EventsFilterByCount,
		NotOperators.NotActivitiesFilterByCount
	],
	[PropertyTypes.Interest]: [CustomFunctionOperators.InterestsFilter],
	[PropertyTypes.Number]: [
		RelationalOperators.EQ,
		RelationalOperators.GE,
		RelationalOperators.GT,
		RelationalOperators.LE,
		RelationalOperators.LT,
		RelationalOperators.NE
	],
	[PropertyTypes.SessionDateTime]: [PropertyTypes.SessionDateTime],
	[PropertyTypes.SessionNumber]: [CustomFunctionOperators.SessionsFilter],
	[PropertyTypes.SessionText]: [CustomFunctionOperators.SessionsFilter],
	[PropertyTypes.Text]: [
		RelationalOperators.EQ,
		RelationalOperators.NE,
		FunctionalOperators.Contains,
		NotOperators.NotContains,
		DisplayOnlyOperators.IsKnown,
		DisplayOnlyOperators.IsUnknown
	]
};

/**
 * Values for criteria row inputs.
 */

export const STRING_OPTIONS = [
	FunctionalOperators.Contains,
	NotOperators.NotContains,
	RelationalOperators.EQ,
	RelationalOperators.NE
];

export const STRING_OPERATOR_LABELS_MAP = {
	[FunctionalOperators.Contains]: Liferay.Language.get('contains-fragment'),
	[NotOperators.NotContains]: Liferay.Language.get('not-contains-fragment'),
	[RelationalOperators.EQ]: Liferay.Language.get('is-fragment'),
	[RelationalOperators.NE]: Liferay.Language.get('is-not-fragment')
};

export const BOOLEAN_OPTIONS = [
	{
		label: Liferay.Language.get('true'),
		value: 'true'
	},
	{
		label: Liferay.Language.get('false'),
		value: 'false'
	}
];

export const INTEREST_BOOLEAN_OPTIONS = [
	{
		label: Liferay.Language.get('is-fragment'),
		value: 'true'
	},
	{
		label: Liferay.Language.get('is-not-fragment'),
		value: 'false'
	}
];

export const OCCURENCE_OPTIONS = [
	{
		key: RelationalOperators.GE,
		label: Liferay.Language.get('at-least-fragment'),
		value: RelationalOperators.GE
	},
	{
		key: RelationalOperators.LE,
		label: Liferay.Language.get('at-most-fragment'),
		value: RelationalOperators.LE
	}
];

export const GEOLOCATION_OPTIONS = [
	{
		label: Liferay.Language.get('was-fragment'),
		value: RelationalOperators.EQ
	},
	{
		label: Liferay.Language.get('was-not-fragment'),
		value: RelationalOperators.NE
	},
	{
		label: Liferay.Language.get('contained-fragment'),
		value: FunctionalOperators.Contains
	},
	{
		label: Liferay.Language.get('did-not-contain-fragment'),
		value: NotOperators.NotContains
	}
];

export const TIME_CONJUNCTION_OPTIONS = [
	{
		label: Liferay.Language.get('since-fragment'),
		value: SINCE
	},
	{
		label: Liferay.Language.get('after-fragment'),
		value: RelationalOperators.GT
	},
	{
		label: Liferay.Language.get('before-fragment'),
		value: RelationalOperators.LT
	},
	{
		label: Liferay.Language.get('between-fragment'),
		value: FunctionalOperators.Between
	},
	{
		label: Liferay.Language.get('ever-fragment'),
		value: EVER
	},
	{
		label: Liferay.Language.get('on-fragment'),
		value: RelationalOperators.EQ
	}
];

export const ACTIVITY_KEY = 'activityKey';
export const EVENT_KEY = 'eventDefinitionId';

export const TIME_PERIOD_OPTIONS = [
	{
		label: Liferay.Language.get('last-24-hours'),
		value: TimeSpans.Last24Hours
	},
	{
		label: Liferay.Language.get('yesterday'),
		value: TimeSpans.Yesterday
	},
	{
		label: Liferay.Language.get('last-seven-days'),
		value: TimeSpans.Last7Days
	},
	{
		label: Liferay.Language.get('last-28-days'),
		value: TimeSpans.Last28Days
	},
	{
		label: Liferay.Language.get('last-30-days'),
		value: TimeSpans.Last30Days
	},
	{
		label: Liferay.Language.get('last-90-days'),
		value: TimeSpans.Last90Days
	}
];

export {TimeSpans};
