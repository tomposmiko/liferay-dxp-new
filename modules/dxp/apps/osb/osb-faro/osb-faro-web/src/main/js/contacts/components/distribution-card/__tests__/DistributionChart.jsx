import DistributionChart from '../DistributionChart';
import mockStore from 'test/mock-store';
import Promise from 'metal-promise';
import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {cleanup, render} from '@testing-library/react';
import {DistributionTab} from 'shared/util/records';
import {FieldContexts} from 'shared/util/constants';
import {Provider} from 'react-redux';

jest.unmock('react-dom');

describe('DistributionCard DistributionChart', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(
			<Provider store={mockStore()}>
				<BrowserRouter>
					<DistributionChart
						distributionKey='individualsDashboard'
						fetchDistribution={() => Promise.reject()}
						selectedTab={
							new DistributionTab({
								context: FieldContexts.Demographics,
								propertyType: 'number',
								title: 'Tab 1'
							})
						}
						viewAllLink='test/:id'
					/>
				</BrowserRouter>
			</Provider>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders empty state via props with noResultsRenderer', () => {
		const {container} = render(
			<Provider store={mockStore()}>
				<BrowserRouter>
					<DistributionChart
						distributionKey='individualsDashboard'
						fetchDistribution={() => Promise.reject()}
						noResultsRenderer={() => <div>{'empty state'}</div>}
						selectedTab={
							new DistributionTab({
								context: FieldContexts.Demographics,
								propertyType: 'number',
								title: 'Tab 1'
							})
						}
						viewAllLink='test/:id'
					/>
				</BrowserRouter>
			</Provider>
		);

		expect(container).toMatchSnapshot();
	});
});
