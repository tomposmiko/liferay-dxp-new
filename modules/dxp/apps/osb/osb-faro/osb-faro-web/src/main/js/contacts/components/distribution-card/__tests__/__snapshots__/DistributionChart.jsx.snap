// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DistributionCard DistributionChart renders 1`] = `
<div>
  <div
    class="card-body align-center"
  >
    <div />
  </div>
  <div
    class="card-footer"
  >
    <a
      class="button-root btn btn-link btn-sm"
      href="/test/:id"
    >
      Explore Breakdown
      <span
        class="inline-item inline-item-after"
      >
        <svg
          class="icon-root lexicon-icon lexicon-icon-angle-right"
          viewBox=""
        >
          <use
            xlink:href="/o/osb-faro-web/dist/sprite.svg#angle-right"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;

exports[`DistributionCard DistributionChart renders empty state via props with noResultsRenderer 1`] = `
<div>
  <div
    class="card-body align-center"
  >
    <div>
      <div>
        empty state
      </div>
    </div>
  </div>
  <div
    class="card-footer"
  >
    <a
      class="button-root btn btn-link btn-sm"
      href="/test/:id"
    >
      Explore Breakdown
      <span
        class="inline-item inline-item-after"
      >
        <svg
          class="icon-root lexicon-icon lexicon-icon-angle-right"
          viewBox=""
        >
          <use
            xlink:href="/o/osb-faro-web/dist/sprite.svg#angle-right"
          />
        </svg>
      </span>
    </a>
  </div>
</div>
`;
